### ISSUE TITLE
recursor 3.7.3 unresponsive after a "rec_control wipe-cache .hosting.absolutok.com"

### ISSUE THREAD

**Issue by jellytoo** [PowerDNS/pdns#4098]
possibly after "rec_control wipe-cache .hosting.absolutok.com"

[14:33] # rec_control wipe-cache hosting.absolutok.com. 
wiped 2 records, 0 negative records
[14:33] # rec_control wipe-cache absolutok.com.  
wiped 2 records, 0 negative records
[14:33] # rec_control wipe-cache .hosting.absolutok.com. <-- possibly here
Fatal: Timeout waiting for answer from control channel
[14:34] # rec_control wipe-cache absolutok.com. 
Fatal: Timeout waiting for answer from control channel

Logs show:

```
[...]
@40000000577a57c3242dc994 Jul 04 14:34:01 1 question answered from packet cache from 10.50.26.253
@40000000577a57c32431c51c Jul 04 14:34:01 1 question answered from packet cache from 10.50.26.253
@40000000577a57c3243595ac Jul 04 14:34:01 1 question answered from packet cache from 10.50.26.253
@40000000577a57c324372fd4 Jul 04 14:34:01 1 question answered from packet cache from 10.50.26.253
@40000000577a57c324d2df4c Jul 04 14:34:01 STL Exception: DNSPacketWriter::xfrLabel() found empty label in the middle of name
@40000000577a5bbf1f93bb8c Jul 04 14:51:01 Refreshed . records
@40000000577a5c3c2867d57c Jul 04 14:53:06 Refreshed . records
@40000000577a5efe242257e4 Jul 04 15:04:52 PowerDNS Recursor 3.7.3  (jenkins@autotest.powerdns.com) (C) 2001-2015 PowerDNS.COM BV  <--- restarted here
@40000000577a5efe24259bd4 Jul 04 15:04:52 Using 64-bits mode. Built on 20150609121259 by root@autotest.powerdns.com, gcc 4.7.2.
[...]

```

Environment: Debian 7.1 amd64, pdns-recursor 3.7.3-1 amd64 package from upstream automated semistatic(?) builds.

**Comment by jellytoo** [comment]
[core.8125.gz](https://github.com/PowerDNS/pdns/files/346374/core.8125.gz)

**Comment by Habbie** [comment]
This is fixed on master/4.x, depending on whether this was before or after DNSName we may be able to just cherrypick the fix.

**Comment by pieterlexis** [comment]
Closed in #4165
